File: src/python/core/domain/enums/skill_level.py
from enum import Enum


class SkillLevel(Enum):
    LOW = "Low"
    MEDIUM = "Medium"
    HIGH = "High"
    VERY_HIGH = "Very High"
    
    @classmethod
    def from_string(cls, value: str) -> 'SkillLevel':
        if not value:
            return None
        
        normalized = value.strip()
        
        mappings = {
            "Low": cls.LOW,
            "Medium": cls.MEDIUM,
            "High": cls.HIGH,
            "Very High": cls.VERY_HIGH,
            "VeryHigh": cls.VERY_HIGH,
        }
        
        return mappings.get(normalized, None)
    
    @classmethod
    def is_strong(cls, level: 'SkillLevel') -> bool:
        return level in [cls.HIGH, cls.VERY_HIGH]

File: src/dotnet/Core/Domain/Enums/SkillLevel.cs
namespace Rag.Candidates.Core.Domain.Enums;

public enum SkillLevel
{
    Low = 0,
    Medium = 1,
    High = 2,
    VeryHigh = 3
}

public static class SkillLevelExtensions
{
    public static SkillLevel? FromString(string value)
    {
        if (string.IsNullOrWhiteSpace(value))
            return null;

        var normalized = value.Trim();

        return normalized switch
        {
            "Low" => SkillLevel.Low,
            "Medium" => SkillLevel.Medium,
            "High" => SkillLevel.High,
            "Very High" => SkillLevel.VeryHigh,
            "VeryHigh" => SkillLevel.VeryHigh,
            _ => null
        };
    }

    public static bool IsStrong(SkillLevel level)
    {
        return level == SkillLevel.High || level == SkillLevel.VeryHigh;
    }

    public static bool IsStrong(string levelString)
    {
        var level = FromString(levelString);
        return level.HasValue && IsStrong(level.Value);
    }
}
